### PR TITLE
[#19123] Public Submenu under Registered parent menu visible in incorrect place

### PR DESCRIPTION
**Pull request by o-samoilov**
Pull Request for Issue [#19123] - Public Submenu under Registered parent menu visible in incorrect place. (https://issues.joomla.org/tracker/joomla-cms/19123)

### Summary of Changes
When menu is rendering we check parent menu access

### Testing Instructions
Create menu 
-Main (Registered)
  --Submenu (Public)

Open page with menu with public access. 

### Actual result
"Main" and "Submenu" will not show.

### CODE REVIEW

**Comment by Quy** [comment]
Please use tabs and not spaces for indentation.

File: libraries/src/Menu/AbstractMenu.php [hunk 1]
```diff
@@ -265,29 +264,51 @@ public function getItems($attributes, $values, $firstonly = false)
...
 			}
 
-			$test = true;
+			$isNeedAddItem = true;
 
-			for ($i = 0; $i < $count; $i++)
+			for ($i = 0; $i < count($attributes); $i++)
 			{
+				// Check if parent item has access
+				if ($attributes[$i] == 'access') {
```
Comment (Quy) [inline libraries/src/Menu/AbstractMenu.php]: Move `{` to its own line. Repeat below. Please read [Joomla Coding Standards](https://developer.joomla.org/coding-standards/php-code.html).

File: libraries/src/Menu/AbstractMenu.php [hunk 2]
```diff
@@ -265,29 +264,54 @@ public function getItems($attributes, $values, $firstonly = false)
...
+					$parentItem = $this->getItem($item->parent_id);
+
+					while (!is_null($parentItem))
+					{
+						if (!in_array($parentItem->{$attributes[$i]}, $values[$i]))
+						{
+							$isNeedAddItem = false;
+							break;
+						}
+						else
```
Comment (Quy) [inline libraries/src/Menu/AbstractMenu.php]: Combine lines 283-285 to make it be `elseif` to reduce one level of indentation. Or just make it another `if` without the `else`.

Comment (o-samoilov) [reply libraries/src/Menu/AbstractMenu.php]: Hi, thank you for help. I've installed phpcs like in [guide](https://docs.joomla.org/Joomla_CodeSniffer) and configurate PHPStorm. I use "squizlabs/php_codesniffer": "~2.8" and "joomla/coding-standards": "~2.0@alpha"

But when I check file what I've changed "libraries/src/Menu/AbstractMenu.php" I see [next ](http://prntscr.com/n1lny0)
I have not any exeption about PHPDocs, where might be a mistake?

File: libraries/src/Menu/AbstractMenu.php [hunk 3]
```diff
@@ -244,19 +244,17 @@ public function getActive()
 	 *
 	 * @param   mixed    $attributes  The field name(s).
 	 * @param   mixed    $values      The value(s) of the field. If an array, need to match field names
-	 *                                each attribute may have multiple values to lookup for.
-	 * @param   boolean  $firstonly   If true, only returns the first item found
+	 *                               each attribute may have multiple values to lookup for.
```
Comment (Quy) [inline libraries/src/Menu/AbstractMenu.php]:
```suggestion
	 *                                each attribute may have multiple values to lookup for.
```

**Comment by ghost** [comment]
@chetanmadaan @barrycox please test > https://docs.joomla.org/Testing_Joomla!_patches

**Comment by ghost** [comment]
reminder for @chetanmadaan @barrycox to test as you reported at #19123 as really major bug.

**Comment by ghost** [comment]
After click on ```Apply Patch```:

Screenshot_2020-09-09 Joomla Patch Tester - fw3 - Administration

**Comment by zero-24** [comment]
Dear @a-samoylov 

in preperation of the upcomming release of Joomla 3.10 we have used GitHubs rename feature to rename the staging branch into 3.10-dev. Usually GitHub moves all existing PRs towards the new branch just fine, but here it didnt work. The reason seems to be that the fork of the CMS that was used as base for this PR has been deleted so GitHub does no longer have a base to rebase the PR against the new branch and we are also not able to reopen the PR. For that reason GitHub closed this PR in my name, when this issue is still valid It would require a new PR against the new 3.10-dev or 4.0-dev branch.

**Pull request closed without merging**